### PR TITLE
add var for theming images , advanced options

### PR DESCRIPTION
**Pull request by compagnon**
Signed-off-by: Guillaume COMPAGNON <gcompagnon@outlook.com>

could be valuable to set 2 scss variables for theming images as already done for the logo and the background

@skjnldsv ?

### CODE REVIEW

**Comment by skjnldsv** [comment]
Sure thing!
@juliushaertl ? :)

**Review by ChristophWurst — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Tests are failing

```



1) OCA\Theming\Tests\CapabilitiesTest::testGetCapabilities with data set #0 ('name', 'url', 'slogan', '#FFFFFF', '#000000', 'logo', 'background', 'http://absolute/', true, array('name', 'url', 'slogan', '#FFFFFF', '#000000', '#aaaaaa', 'http://absolute/logo', 'http://absolute/background', false, false))
101 | OCA\Theming\ThemingDefaults::getLogo(true): string was not expected to be called more than once.
102 |  
103 | /drone/src/apps/theming/lib/Capabilities.php:86
104 | /drone/src/apps/theming/tests/CapabilitiesTest.php:185
105 |  
106 | 2) OCA\Theming\Tests\CapabilitiesTest::testGetCapabilities with data set #1 ('name1', 'url2', 'slogan3', '#01e4a0', '#ffffff', 'logo5', 'background6', 'http://localhost/', false, array('name1', 'url2', 'slogan3', '#01e4a0', '#ffffff', '#01e4a0', 'http://localhost/logo5', 'http://localhost/background6', false, true))
107 | OCA\Theming\ThemingDefaults::getLogo(true): string was not expected to be called more than once.
108 |  
109 | /drone/src/apps/theming/lib/Capabilities.php:86
110 | /drone/src/apps/theming/tests/CapabilitiesTest.php:185
111 |  
112 | 3) OCA\Theming\Tests\CapabilitiesTest::testGetCapabilities with data set #2 ('name1', 'url2', 'slogan3', '#000000', '#ffffff', 'logo5', 'backgroundColor', 'http://localhost/', true, array('name1', 'url2', 'slogan3', '#000000', '#ffffff', '#000000', 'http://localhost/logo5', '#000000', true, false))
113 | OCA\Theming\ThemingDefaults::getLogo(true): string was not expected to be called more than once.
114 |  
115 | /drone/src/apps/theming/lib/Capabilities.php:86
116 | /drone/src/apps/theming/tests/CapabilitiesTest.php:185
117 |  
118 | 4) OCA\Theming\Tests\CapabilitiesTest::testGetCapabilities with data set #3 ('name1', 'url2', 'slogan3', '#000000', '#ffffff', 'logo5', 'backgroundColor', 'http://localhost/', false, array('name1', 'url2', 'slogan3', '#000000', '#ffffff', '#000000', 'http://localhost/logo5', '#000000', true, true))
119 | OCA\Theming\ThemingDefaults::getLogo(true): string was not expected to be called more than once.
120 |  
121 | /drone/src/apps/theming/lib/Capabilities.php:86
122 | /drone/src/apps/theming/tests/CapabilitiesTest.php:185
123


```

**Comment by compagnon** [comment]
> Tests are failing
>
because of the 3x use of getLogo() to initialize the array of Capabilities of theming (the theming apps)

In the current code , by default, the svg logo image of Nextcloud is used to get the logo ,the favicon and the logoheader

[apps/theming/lib/ImageManager.php#L82-L88](https://github.com/nextcloud/server/blob/b7577b6401cb68a6a1f4dc9090facb79a6438753/apps/theming/lib/ImageManager.php#L82-L88)
```
try {
			$image = $this->getImage($key, $useSvg);
			return $this->urlGenerator->linkToRoute('theming.Theming.getImage', [ 'key' => $key ]) . '?v=' . $cacheBusterCounter;
} catch (NotFoundException $e) {
}
switch ($key) {
			case logo:
			case 'logoheader':
			case 'favicon':
				return $this->urlGenerator->imagePath('core', 'logo/logo.png') . '?v=' . $cacheBusterCounter;
```


what is the choice to do:
- [ ]  add 2 new getter : getLogoHeader($useSvg = true) and getFavicon($useSvg = true) 
in OCA\Theming\ThemingDefaults (and /lib/private/legacy/defaults.php \OC_Defaults ? (with some refactoring in ImageManager to be coherent 

-  [ ] or modify the CapabilitiesTest for being able to call 3x the getLogo ?

**Comment by kesselb** [comment]
@compagnon could you rebase this branch? So the CI can run again.

**Comment by ChristophWurst** [comment]
> @compagnon could you rebase this branch? So the CI can run again.

:ping_pong: :grin:

File: apps/theming/lib/ThemingDefaults.php [hunk 1]
```diff
@@ -285,10 +285,10 @@ public function getScssVariables() {
 			'theming-logoheader-mime' => "'" . $this->config->getAppValue('theming', 'logoheaderMime') . "'",
 			'theming-favicon-mime' => "'" . $this->config->getAppValue('theming', 'faviconMime') . "'"
 		];
-
+
```
Comment (kesselb) [inline apps/theming/lib/ThemingDefaults.php]:
```suggestion

```

**Comment by kesselb** [comment]
Remove some whitespace and Rebased the branch because many test failures.

File: apps/theming/lib/ThemingDefaults.php (hunk 1 above)
Comment (kesselb) [inline apps/theming/lib/ThemingDefaults.php]: Please remove the whitespace

**Comment by kesselb** [comment]
Mind to squash the commits into one?

**Comment by compagnon** [comment]
if you dont mind, 
yes I will squash the commits into just one

**Comment by kesselb** [comment]
cc @skjnldsv @ChristophWurst :ping_pong:

**Review by skjnldsv — APPROVED** [review]
Code is good to me :)

**Review by ChristophWurst — APPROVED** [review]
Looks good!

File: apps/theming/lib/Capabilities.php [hunk 2]
```diff
@@ -83,6 +83,8 @@ public function getCapabilities() {
 					$this->url->getAbsoluteURL($this->theming->getBackground()),
 				'background-plain' => $backgroundLogo === 'backgroundColor',
 				'background-default' => !$this->util->isBackgroundThemed(),
+				'logoheader' => $this->url->getAbsoluteURL($this->theming->getLogo()),
+				'favicon' => $this->url->getAbsoluteURL($this->theming->getLogo()),
```
Comment (ShGKme) [inline apps/theming/lib/Capabilities.php:87]: @skjnldsv @ChristophWurst Any chance you remember, why capabilities for `logheader` and `favicon` were added here with the same `getLogo()`?

**Pull request merged**